scripts/other-yes-list/parse.ts

```diff
@@ -0,0 +1,63 @@
+export type Course = {
+  /** Corresponds to the "prefix" column on the PDF */
+  subject: string;
+  /** Corresponds to the "code" column on the PDF */
+  catalogNbr: number;
+  /** The liberal studies categories satisfied by the course */
+  categories: readonly string[];
+};
+
+// creates list of course entries
+export function genCourseEntries(data) {
+  const courseEntries: Course[] = [];
+  let col = 'course prefix';
+  let subject = '';
+  let catalogNbr = 0;
+  let categories = [];
+
+  // find the index of the first course (right after Notes)
+  let start = 0;
+  for (let i = 0; i < data.length; i += 1) {
+    try {
+      const v = data[i][0].str.trim();
+      if (v.slice(0, 5) === 'Notes') {
+        start = i + 1;
+      }
+    } catch {
+      // in case there is no list of approved courses
+    }
```

Comment:
What exception are we trying to catch here?

Reply:
Not sure but it gives me an error for invalid data and the suggested fix is that I use try/catch.

Reply:
I think we should explicit handle the case that causes an error